src/ddmd/parse.d

```diff
@@ -5112,10 +5112,7 @@ final class Parser(AST) : Lexer
         case TOKsemicolon:
             if (!(flags & PSsemi_ok))
             {
-                if (flags & PSsemi)
-                    warning(loc, "use '{ }' for an empty statement, not a ';'");
-                else
-                    error("use '{ }' for an empty statement, not a ';'");
+                error("use '{ }' for an empty statement, not a ';'");
```

Comment:
You cant pass from warning to error. There must be a  deprecation period before.